docs/code-review.md

```diff
@@ -73,11 +75,11 @@ You should approve a pull request when you have confirmed that:
 2. it doesn't introduce new defects or code that is hard to maintain
 3. all new and modified code has thorough unit test coverage
 4. integration tests have been added where appropriate
-3. there are no unanswered questions or comments against the pull request
+5. there are no unresolved questions or comments against the pull request
 
 Occasionally, it may be prudent to accept a pull request which does not meet all of the above requirements, such as to resolve an urgent issue with the live product. Such cases must always be agreed between the product owner, author(s) and reviewer(s).
 
-If comments or questions on a pull request have been addressed elsewhere (e.g. face-to-face or on Slack), ensure that the outcome is recorded in replies to each comments so that it is visible to anyone looking back at the pull request in future for information.
+If comments or questions on a pull request have been addressed elsewhere (e.g. face-to-face or on Slack), ensure that the outcome is recorded in comment replies so that it is visible to anyone looking back at the pull request in future for information. Use the Resolve button to make it clear which comments have been addressed and which still need attention. A pull request should only be approved after all comments have been resolved.
```

Comment:
Should we clarify who should resolve the conversation?  The writer or the reviewer.

Reply:
That's a great point.

My feeling is that it should be the reviewer so that they explicitly confirm that their concerns have been addressed. It seems too easy for an author to misunderstand a comment and think they've addressed it when they haven't. Sometimes I have resolved comments against my own PRs - when trying to work out what still needed looking at - but it always makes me uncomfortable because of that risk of misunderstanding.

It gets a bit more tricky to know who should resolve it when several reviewers have replied to the same comment thread. Another problem that could occur is a reviewer going on holiday or off sick after leaving comments. I wouldn't want a PR to be held up because the reviewer isn't there to click "Resolve". Perhaps then it should be open to a follow-up reviewer to decide whether the comments have been addressed and resolve them accordingly?

Reply:
I also feel it should be the reviewer but you make a good point about multiple or absent reviewers.

For multiple reviewers, in most cases the conversation will be about the same topic so any of them could resolve as long as the conversation had met a natural conclusion and agreement of what to do and if that involves a change, has that change been made.  But I think it's a judgement call we'd be capable of making.

We could have a follow up reviewer but then that does risk the reviewer also misunderstands the original reviewer.  Do you think we should close it as unresolved so we're not speaking on behalf of the absent reviewer?

Reply:
Yeah, I'm not sure about my idea of a follow-up reviewer hitting resolve either. Sadly, there isn't an option to close a conversation as unresolved, but we could say that it's OK for a follow-up reviewer to approve the pull request with unresolved comments if they think those comments have been addressed and their author is unavailable to confirm.

Reply:
Yeah that's what I meant, close the pull request leaving the comment unresolved.  Should have said "approve" not close.